### PR TITLE
feat: Add support for `maintenance_options`

### LINKED ISSUE #310: instance auto_recovery option to the EC2 instance
## Is your request related to a new offering from AWS?
Is this functionality available in the [AWS provider for Terraform](https://registry.terraform.io/providers/hashicorp/aws/latest/docs)? See [CHANGELOG.md](https://github.com/hashicorp/terraform-provider-aws/blob/main/CHANGELOG.md), too.
  - Yes ✅: please list the AWS provider version which introduced this functionality
  - This change is supported in [ver 4.12.0](https://registry.terraform.io/providers/hashicorp/aws/4.12.0/docs/resources/instance#auto_recovery) through [latest version](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#auto_recovery)

## Is your request related to a problem? Please describe.

AWS EC2 instance has added automatic recovery option function.
Since this value cannot be changed in the current module, I would like to ask you to modify the module so that it can be changed.

## Describe the solution you'd like.

I would like to include the optional input parameter [auto_recovery](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/instance#auto_recovery)

### PR DESCRIPTION
**Pull request by magreenbaum**
## Description
Adding `auto_recovery` option to toggle between `default` and `disabled`. This argument is computed if not configured. 
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/ec2-instance-recover.html 

## Motivation and Context
- Resolves #310

## Breaking Changes
No.

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -127,6 +127,13 @@ resource "aws_instance" "this" {
     }
   }
 
+  dynamic "maintenance_options" {
```
Comment (magreenbaum) [inline main.tf:130]: Ran across [this PR](https://github.com/terraform-aws-modules/terraform-aws-autoscaling/pull/193) and this implementation made more sense.

**Review by bryantbiggs — APPROVED** [review]
Thank you!

**Comment by bryantbiggs** [comment]
Ugh, GitHub changed something with the rate limiting and now our actions are failing early. Give me a sec to fix these with a token - changes look good and we'll be able to merge straight away

**Comment by antonbabenko** [comment]
This PR is included in version 4.3.0 :tada:

**Pull request merged**